envoy-control-core/src/test/java/pl/allegro/tech/servicemesh/envoycontrol/v2/SimpleCacheTest.java

```diff
@@ -519,8 +521,13 @@ protected static void assertThatWatchReceivesSnapshot(WatchAndTracker watchAndTr
 
         assertThat(response).isNotNull();
         assertThat(response.version()).isEqualTo(snapshot.version(watchAndTracker.watch.request().getTypeUrl()));
-        assertThat(response.resources().toArray(new Message[0]))
-                .containsExactlyElementsOf(snapshot.resources(watchAndTracker.watch.request().getTypeUrl()).values());
+        Message[] responseValues = response.resources().toArray(new Message[0]);
```

Comment:
I do not know why but the `containsExactlyElementsOf` stopped working :/ even though the values underneath are the same. I converted this to an equivalent assertion but please pay attention here.

Reply:
I think debugging using breakpoint, line by linem will tell us where where the problem is.

Reply:
Maybe the lack of `toArray` for snapshot.resources is the problem?